esp/src/src-react/components/UserGroups.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from "react";
+import { CommandBar, ContextualMenuItemType, ICommandBarItemProps } from "@fluentui/react";
+import { useConst } from "@fluentui/react-hooks";
+import { scopedLogger } from "@hpcc-js/util";
+import * as Observable from "dojo/store/Observable";
+import { Memory } from "src/Memory";
+import * as WsAccess from "src/ws_access";
+import nlsHPCC from "src/nlsHPCC";
+import { ShortVerticalDivider } from "./Common";
+import { pushUrl } from "src-react/util/history";
+import { HolyGrail } from "src-react/layouts/HolyGrail";
+import { DojoGrid, selector } from "./DojoGrid";
+import { UserAddGroupForm } from "./forms/UserAddGroup";
+
+const logger = scopedLogger("../components/UserGroups.tsx");
+
+const defaultUIState = {
+    hasSelection: false,
+};
+
+interface UserGroupsProps {
+    username?: string;
+}
+
+export const UserGroups: React.FunctionComponent<UserGroupsProps> = ({
+    username,
+}) => {
+
+    const [grid, setGrid] = React.useState<any>(undefined);
+    const [selection, setSelection] = React.useState([]);
+    const [showAdd, setShowAdd] = React.useState(false);
+    const [uiState, setUIState] = React.useState({ ...defaultUIState });
+
+    //  Grid ---
+    const gridStore = useConst(new Observable(new Memory("name")));
+    const gridSort = useConst([{ attribute: "name", "descending": false }]);
+    const gridQuery = useConst({});
+    const gridColumns = useConst({
+        check: selector({ width: 27, label: " " }, "checkbox"),
+        name: {
+            label: nlsHPCC.GroupName,
+            formatter: function (_name, idx) {
+                return `<a href="#/security/groups/${_name}">${_name}</a>`;
```

Comment:
`_name` should be run through a sanitizer!!!

Reply:
There might be a lot of these across the codebase. Changed this one instance to use just a simple regex to sanitize, but should maybe find a good / robust library for this? Dompurify?

Reply:
Open a ticket and we can investigate which lib will work for us.

Reply:
FYI - all React based items are sanitized by default...